NewsFeed/views/CCBRNewsViewController.m

```diff
@@ -185,11 +226,21 @@ - (CGSize)collectionView:(UICollectionView *)collectionView layout:(UICollection
     return CGSizeMake(itemWidth, itemHeight);
 }
 
+- (void)scrollViewDidScroll:(UIScrollView *)scrollView {
+    CGFloat visibleHeight = scrollView.frame.size.height - scrollView.contentInset.top - scrollView.contentInset.bottom;
+    CGFloat y = scrollView.contentOffset.y + scrollView.contentInset.top + scrollView.contentInset.bottom;
+    CGFloat offset = 300.0;
+    CGFloat threshhold = MAX(offset, scrollView.contentSize.height - visibleHeight - offset);
+    if (y >= threshhold) {
+        [self.viewModel loadMore];
+    }
```

Comment:
Can you please move this logic to the view-model?
The view-controller should reflect the state in the view-model, it should not "know" the logic or "decide" when to load more.
If you put the logic of "loading more" in the view-controller, it will be hard to unit-test logic. Because you will need to load the view, fill the data and wait for the willDisplayCell method to be called!